lib/hanami/providers/db/sql_adapter.rb

```diff
@@ -85,7 +85,7 @@ def configure_for_database(database_url)
 
         # @api private
         def gateway_options
-          {extensions: config.extensions}
+          {extensions: extensions}
```

Comment:
This eliminates an edge-case where you would end up with `{extensions: nil}` if you didn't set anything. Now the type signature of this value should always be consistent.